usdt-impl/src/common.rs

```diff
@@ -208,8 +208,10 @@ pub(crate) fn build_probe_macro(
                 compile_error!("USDT probe macros should be invoked with a closure returning the arguments");
             };
             ($args_lambda:expr) => {
-                #type_check_block
-                #impl_block
+                {
+                    #type_check_block
+                    #impl_block
+                }
```

Comment:
I think it's not, you're right. The scoping around the `type_check_block` _is_ useful, since it prevents the `use` statements we insert from polluting the scope of the `impl_block`. Those are both in the macro, and it looks to the user like there are some magical import statements. Which there _are_, but I don't think we want them to know that :)